### PR TITLE
[editorial] Add Hugo front matter to selected pages

### PR DESCRIPTION
**Pull request by chalin**
- Contributes to https://github.com/open-telemetry/opentelemetry.io/issues/1467

/cc @austinlparker

### CODE REVIEW

File: specification/overview.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+<!---
+weight: 1
+--->
+
```
Comment (arminru) [inline specification/overview.md:4]: Would it be possible to add a comment that this is needed for the website? I could imagine someone passing by in a few months thinking this is some leftover/accidental comment and then removing it.

**Review by arminru — APPROVED** [review]

File: specification/overview.md (hunk 1 above)
Comment (chalin) [reply specification/overview.md:4]: Sure, how about something like this:

Comment (arminru) [reply specification/overview.md:4]: @chalin Yes I think this would be helpful!

Comment (arminru) [reply specification/overview.md:4]: Sure!

Comment (chalin) [reply specification/overview.md:4]: - Done via #2651

**Pull request merged**